src/main/java/com/orbit/frc/subsystems/leds/Leds.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Package: com.orbit.frc.subsystems.leds
+ * Description: This package contains classes related to controlling LEDs on a robot.
+ */
+
+package com.orbit.frc.subsystems.leds;
+
+import java.util.HashMap;
+
+import edu.wpi.first.wpilibj.Spark;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/*
+ * Class: Leds
+ * Description: This class represents the subsystem responsible for controlling LEDs on a robot.
+ */
+public class Leds {
+    private final LedConfig config; // Configuration object for LEDs
+
+    private double LEDColour; // Current color of LEDs
+    public LEDStates LEDstate; // Current state of LEDs
+
+    HashMap<LEDStates, Double> LEDStatesMap = new HashMap<>(); // Map to store LED states and corresponding colors
+
+    /*
+     * Constructor: Leds
+     * Description: Initializes the LED subsystem with the provided configuration.
+     * Parameters:
+     *  - config: Configuration object for LEDs
+     */
+    public Leds(LedConfig config) {
+        this.LEDController = new Spark(config.LEDPort); // Initialize LED controller
```

Comment:
remember to change references to the controller when its in the config, and also check to see if it should be inited from port or just using the object passed in the config

Reply:
imo it should be inited from the port, makes it less confusing in config

Reply:
If the user wants to provide their own spark object, we should support that